newIDE/app/src/AssetStore/PrivateAssets/AuthorizedAssetImage.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import * as React from 'react';
+import PrivateAssetsAuthorizationContext from './PrivateAssetsAuthorizationContext';
+import PlaceholderLoader from '../../UI/PlaceholderLoader';
+import { createAuthorizedUrl } from '../../Utils/GDevelopServices/Shop';
+
+type Props = {|
+  url: string,
+  style?: Object,
+  alt: string,
+  onError?: () => void,
+  onLoad?: (e: any) => void,
+  hideLoader?: boolean,
+|};
+
+const AuthorizedAssetImage = ({
+  url,
+  style,
+  alt,
+  onError,
+  onLoad,
+  hideLoader,
+}: Props) => {
+  const [authorizedUrl, setAuthorizedUrl] = React.useState(null);
+  const [isImageLoaded, setIsImageLoaded] = React.useState(false);
+  const { authorizationToken, updateAuthorizationToken } = React.useContext(
+    PrivateAssetsAuthorizationContext
+  );
+  const [tries, setTries] = React.useState(0);
```

Comment:
I feel like this could be a ref to avoid trigger rendering

Reply:
yep, fixing this